src/slack/utils.bal

```diff
@@ -0,0 +1,436 @@
+        if (ch.name == channelName) {
+            return (ch.id).toString();
+        }
+    }
+    return error(SLACK_ERROR_CODE, message = "Channel " + channelName + " does not exist");
+}
+
+function archiveConversation(http:Client slackClient, string channelId) returns @tainted error? {
+    string url = ARCHIVE_CHANNEL_PATH + channelId;
+    return handleArchiveResponse(slackClient, url);
+}
+
+function unArchiveConversation(http:Client slackClient, string channelId) returns @tainted error? {
+    string url = UNARCHIVE_CHANNEL_PATH + channelId;
+    return handleArchiveResponse(slackClient, url);
+}
+
+function handleArchiveResponse(http:Client slackClient, string url) returns @tainted error? {
+    http:Response|error response = slackClient->post(url, EMPTY_STRING);
+    if (response is error) {
+        return setResError(response);
+    }
+    http:Response resp = <http:Response> response;
+    var jsonResponse = resp.getJsonPayload();
+    if (jsonResponse is error) {
+        return setJsonResError(jsonResponse);
+    }
+    json jsonResp = <json> jsonResponse;
+    var checkOk = check checkOk(jsonResp);       
+}
+
+function getUserIds(http:Client slackClient, string[] users) returns @tainted string|error {
+    string usersList = EMPTY_STRING;
+    http:Response|error response = slackClient->get(LIST_USERS_PATH);
+    if (response is error) {
+        return setResError(response);
+    }
+    http:Response resp = <http:Response> response;
+    var payload = resp.getJsonPayload();
+    if (payload is error) {
+        return setJsonResError(payload);
+    }
+    map<json> jsonPayload = <map<json>> payload;
+    var checkOk = check checkOk(jsonPayload);
+    var members = check jsonPayload.members;
+    json[] memberList = <json[]> members;
+    foreach var user in users {
+        foreach var member in memberList {
+            if (member.name == user) {
+                usersList = usersList + (member.id).toString() + ",";
+            }
+        }
+    } 
+    if (usersList == EMPTY_STRING) {
+        return error(SLACK_ERROR_CODE, message = "Unable to find user ids of the given users");
+    }
+    return usersList;
+}
+
+function getUserId(http:Client slackClient, string user) returns @tainted string|error {
+    string userId = EMPTY_STRING;
+    http:Response|error response = slackClient->get(LIST_USERS_PATH);
+    if (response is error) {
+        return setResError(response);
+    }
+    http:Response resp = <http:Response> response;
+    var payload = resp.getJsonPayload();
+    if (payload is error) {
+        return setJsonResError(payload);
+    }
+    map<json> jsonPayload = <map<json>> payload;
+    var checkOk = check checkOk(jsonPayload);
+    var members = check jsonPayload.members;
+    json[] memArr = <json[]> members;
+    foreach var member in memArr {
+        if (member.name == user) {
+            userId =  (member.id).toString();
+            break;
+        }
+    } 
+    if (userId == EMPTY_STRING) {
```

Comment:
What if we return the `userId` instead of `break` the loop, when we have a match? 

That way we can remove the `userId` variable plus this `if` check, and just return the error. WDYT?